src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,160 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        final Node<T> oldTail = tail;
+        Node<T> newNode = new Node<>(oldTail, value, null);
+        tail = newNode;
+        if (oldTail == null) {
+            head = newNode;
+        } else {
+            oldTail.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndexForAdd(index);
+
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> successor = node(index);
+            Node<T> predecessor = successor.prev;
+            Node<T> newNode = new Node<>(predecessor, value, successor);
+            successor.prev = newNode;
+            if (predecessor == null) {
+                head = newNode;
+            } else {
+                predecessor.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkPositionIndexForAccess(index);
+        return node(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkPositionIndexForAccess(index);
+        Node<T> current = node(index);
+        T oldValue = current.value;
+        current.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkPositionIndexForAccess(index);
+        return unlink(node(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> element = head; element != null; element = element.next) {
+            if (object == element.value
+                    || (object != null
+                    && object.equals(element.value))) {
+                unlink(element);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> node(int index) {
```

Comment:
Bad naming - method should describe the action

```suggestion
    private Node<T> findNodeByIndex(int index) {
```